src/geom/bounding_box.C

```diff
@@ -62,6 +62,62 @@ bool BoundingBox::contains_point (const Point & p) const
 }
 
 
+
+Real BoundingBox::max_size() const
+{
+  Real size = 0;
+  for (unsigned int d = 0; d != LIBMESH_DIM; ++d)
+    {
+      Real sized = this->second(d) - this->first(d);
+      if (!libmesh_isinf(sized))
+        size = std::max(size, sized);
+    }
+  return size;
+}
+
+
+
+bool BoundingBox::contains_point
+  (const Point & p, Real abs_tol, Real rel_tol) const
```

Comment:
```suggestion
  (const Point & p, const Real abs_tol, const Real rel_tol) const
```

Reply:
We generally don't pass parameters by `const` value.  `const` on a reference parameter is making a promise to the user, but `const` on a value parameter means nothing to the API semantics (because the value is a copy or a temporary anyway depending on input, and in neither case will the calling code ever see it) but can require future changes to the implementation to change the API syntax.

Reply:
You can leave it non-const in the header declaration for the reason you stated but mark it as `const` in the implementation as that helps communicate to the reader that the parameters (at least in the current implementation) do not change